Update nix flake. Bump hyprland v0.42.0 -> v.0.45.2

MattKang opened the pull request:
Update nix flake and lock to bump Hyprland v0.42.0 -> v0.45.2

Addresses #80 but still need ultimate fix to #48. Not entirely sure how one can know what version of Hyprland this project is being built against at any point in time.

dawsers commented:
Thank you for doing this. I want to merge it, but before I have a question, because I know very little about NixOS.

In the beginning, we used to have to provide these flakes, and several people contributed to them because I don't use NixOS. It was painful, because every time I committed some code, someone had to go and verify NixOS was getting the update.

Then, it seems *hyprscroller* was added as an official unstable package to NixOS [here](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=hyprlandPlugins.hyprscroller).

So my question is, do you think it is worth keeping the local flakes in this repo, with the possible confusion it may create, or should redirect everyone to the official package? My point is, I am the maintainer here, and I have no idea about flakes. When someone wants to merge something, I basically cannot judge if what they did is good or maybe broke or simplified some previous effort. Instead, if everyone who wanted to, contributed directly to the NixOS unstable package, the changes would be judged by someone who actually know what they are doing :-), and at the same time there would just be one "ground truth".

MattKang commented:
@dawsers yes I think there's a concern if you are the (sole?) maintainer cannot review the quality of this component. I will personally admit that I didn't read the README, saw there was a flake, and then tried to use it.

Disclaimer: I'm not a `nix` expert. I just use it.

I can think of a few solutions:
1. Drop `nix flake` support. The `nix` community is strong, and either way, they will continue to maintain this.
2. Build a `nix` CI/CD pipeline to verify that everything works (so you don't have to). Will still require maintenance at some point. Note that this could also be used for automatic update in `nixpkgs`.
3. Make a `nix` branch that is updated separately and rebased off of `master`. That way it can at least remain stable even if it lags `master`.

All of these will **require a versioning system** to make it clear what version of Hyprland you're building hyprscroller against at any given commit. Personally, I was so invested in getting the latest hyprscroller that I didn't want to wait for the main `nixpkgs` update. So I just pulled the latest Hyprland flake and updated your hyprscroller flake. I might be a niche case, though.

dawsers commented:
I agree with you. I think I will just remove the flakes and redirect everyone to the NixOS package. Maintenance there will be better. If someone wants to use the latest and NixOS lags a bit, `hyprpm` should still work on NixOS. By previous interactions, it seemed to me NixOS people don't like `hyprpm`, but the flakes here are usually older than the nixOS package from what I have seen.

Thank you for your help and the discussion. It doesn't make sense to merge the PR because I am going to delete the files in a second.
